src/routes/files/handlers.js

```diff
@@ -502,20 +502,32 @@ async function deleteActivityReportObjectiveFile(req, res) {
       return;
     }
 
-    await ActivityReportObjectiveFile.destroy({
+    // Get ARO files to delete (destroy does NOT support join's).
+    const aroFilesToDelete = await ActivityReportObjectiveFile.findAll({
       where: {
         fileId: parseInt(fileId, DECIMAL_BASE),
       },
       include: [
         {
+          as: 'activityReportObjective',
           model: ActivityReportObjective,
           where: {
             activityReportId: parseInt(reportId, DECIMAL_BASE),
-            objectiveIds,
+            objectiveId: objectiveIds,
           },
           required: true,
         },
       ],
+    });
+
+    // Get ARO file ids.
+    const aroFileIds = aroFilesToDelete.map((arof) => arof.id);
```

Comment:
These two code blocks could be merged. add "attributes: ['id'], and raw: true," at the top level and attributes: [] to the include to prevent it from adding any data. This will then return the same array as the map line on 524, so it would not be needed.